### DISCUSSION TITLE
Add a new input mode to the touch device

### DISCUSSION THREAD

**Discussion by ReBiliBin** [ppy/osu#30575] [Ideas]
When using my mobile device to play osu lazer, I found that the current touch input was not very useful, so I thought about adding a new touch input mode to osu lazer, similar to the trackpad on a laptop, dividing the playing area into a moving area and a clicking area just like this:
![Left-Right](https://github.com/user-attachments/assets/e47a2564-4adb-46de-86ce-1d0c3252f660)
or
![Up-Down](https://github.com/user-attachments/assets/e7a8cdbf-d8fd-4845-98c2-98e4377590c8)

**Comment by bdach** [comment]
You can already do this if you keep one finger pressed to the screen to point and click using the others, though? It's just not separated into "areas".

**Reply by ReBiliBin** [reply]
Now it is easy for touch control to mistakenly identify the clicked finger as a moving finger, which can be avoided by separating a single click area. In theory, the "touchpad mode" I think can support input acceleration, so that you can separate the control like using a keyboard and mouse, without worrying about the wrong recognition of your fingers.

**Comment by SupDos** [comment]
You can already do this using screen scaling

If you set it to only "Gameplay", like so:
![osu_2024-11-11_20-44-22](https://github.com/user-attachments/assets/cc5fb552-1588-4df1-b3a2-13846a5352f5)

then the left side becomes tapping only, and the right side becomes movement only

**Reply by shiumano** [reply]
It will not.
If you tap off-screen, the cursor will just go off-screen.

**Reply by SupDos** [reply]
I'm not sure I understand your playstyle then, are you not continuously holding your finger down to move the cursor? Moving the gameplay to the side means that your tapping fingers won't ever get a circle near them, so the game won't have a chance to confuse the taps as you trying to tap one of the circles
